security/pkg/nodeagent/cache/helper.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/gogo/status"
+	"google.golang.org/grpc/codes"
+
+	"istio.io/istio/security/pkg/nodeagent/model"
+)
+
+// SecretExist checks if secret already existed.
+// This API is used for sds server to check if coming request is ack request.
+func (sc *SecretCache) SecretExist(connectionID, resourceName, token, version string) bool {
```

Comment:
better to keep member funcs in original secretcache.go and just move helper functions(like parseCertAndGetExpiryTimestamp) into helper file

Reply:
Makes sense. Will change. Thanks!